### PR TITLE
support new standardize_parameters

### PR DESCRIPTION
**Pull request by strengejacke**

### CODE REVIEW

**Comment by mattansb** [comment]
@strengejacke if you approve, I will do the same for `.extract_parameters_mixed()`.

Note that for some reason the column name remains "Coefficient" - I couldn't find how to change it...


``` r
model <- lm(Sepal.Length ~ Petal.Length + Petal.Width, data = iris)

parameters::model_parameters(model, standardize = "refit")
#> Parameter    | Coefficient |   SE |         95% CI |    t(147) |      p
#> -----------------------------------------------------------------------
#> (Intercept)  |   -1.92e-16 | 0.04 | [-0.08,  0.08] | -4.84e-15 | > .999
#> Petal.Length |        1.15 | 0.15 | [ 0.86,  1.45] |      7.82 | < .001
#> Petal.Width  |       -0.29 | 0.15 | [-0.59,  0.00] |     -1.99 | 0.048

parameters::model_parameters(model, standardize = "basic")
#> Parameter    | Coefficient |   SE |         95% CI | t(147) |      p
#> --------------------------------------------------------------------
#> (Intercept)  |        0.00 | 0.00 | [ 0.00,  0.00] |  43.18 | < .001
#> Petal.Length |        1.15 | 0.15 | [ 0.86,  1.45] |   7.82 | < .001
#> Petal.Width  |       -0.29 | 0.15 | [-0.59,  0.00] |  -1.99 | 0.048
```

<sup>Created on 2020-10-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mattansb** [comment]
(need dev version of effectsize for this to work)

**Comment by strengejacke** [comment]
The column names are only changed in the `print()` method, by calling `parameters_table()`. _effectsize_ should return a column name `"Std_Coefficient"`, or you must rename it manually here.

File: R/extract_parameters.R [hunk 1]
```diff
@@ -229,6 +206,26 @@
...
+
+  if (!is.null(standardize)) {
+    # give minimal attributes required for standardization
+    temp_pars <- parameters
+    class(temp_pars) <- c("parameters_model", class(temp_pars))
+    attr(temp_pars, "ci") <- ci
+    attr(temp_pars, "obj_name") <- model # pass the model as is (this is a cheat - teehee!)
+
+    std_parms <- effectsize::standardize_parameters(temp_pars, method = standardize)
+    parameters$Coefficient <- std_parms$Std_Coefficient
```
Comment (strengejacke) [inline R/extract_parameters.R:220]: You asked why the column is named "Coefficient", but you explicitely did this here?

File: R/extract_parameters.R [hunk 2]
```diff
@@ -229,6 +206,26 @@
...
+    parameters$SE <- attr(std_parms, "standard_error")
+
+    if (!is.null(ci)) {
+      parameters$CI_low <- std_parms$CI_low
```
Comment (strengejacke) [inline R/extract_parameters.R:224]: Should be "Std_CI_low" as name (and for high as well)

File: R/extract_parameters.R (hunk 1 above)
Comment (mattansb) [reply R/extract_parameters.R]: Fixed!

File: R/extract_parameters.R (hunk 2 above)
Comment (mattansb) [reply R/extract_parameters.R:218]: These were previously just `CI_low` / `CI_high`, without the `Std_` prefix.

**Comment by mattansb** [comment]
Okay!

``` r
model <- lm(Sepal.Length ~ Petal.Length + Petal.Width, data = iris)

parameters::model_parameters(model, standardize = "refit")
#> Parameter    | Coefficient |   SE |         95% CI |    t(147) |      p
#> -----------------------------------------------------------------------
#> (Intercept)  |   -1.92e-16 | 0.04 | [-0.08,  0.08] | -4.84e-15 | > .999
#> Petal.Length |        1.15 | 0.15 | [ 0.86,  1.45] |      7.82 | < .001
#> Petal.Width  |       -0.29 | 0.15 | [-0.59,  0.00] |     -1.99 | 0.048

parameters::model_parameters(model, standardize = "basic")
#> Parameter    | Std. Coef. |   SE |         95% CI | t(147) |      p
#> -------------------------------------------------------------------
#> (Intercept)  |       0.00 | 0.00 | [ 0.00,  0.00] |  43.18 | < .001
#> Petal.Length |       1.15 | 0.15 | [ 0.86,  1.45] |   7.82 | < .001
#> Petal.Width  |      -0.29 | 0.15 | [-0.59,  0.00] |  -1.99 | 0.048
```

<sup>Created on 2020-10-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>


So what do you think?

**Review by strengejacke — COMMENTED** [review]
As far as I can see, looks good!

**Comment by strengejacke** [comment]
So what is the order of package submission for the next update-round?

1. insight
2. effectsize
3. parameters

?

**Comment by mattansb** [comment]
Yes, that would be the order.

**Comment by strengejacke** [comment]
Ok, will submit insight next week, or are there any things required to make effectsize work?

**Comment by mattansb** [comment]
And for mixed:

``` r
library(parameters)
library(lme4)
#> Loading required package: Matrix

data(mtcars)
model <- lmer(mpg ~ wt + (1 | gear), data = mtcars)

model_parameters(model, standardize = "refit")
#> Parameter   | Coefficient |   SE |         95% CI | t(28) |      p
#> ------------------------------------------------------------------
#> (Intercept) |       -0.02 | 0.15 | [-0.32,  0.28] | -0.14 | 0.886 
#> wt          |       -0.82 | 0.10 | [-1.02, -0.62] | -7.89 | < .001

model_parameters(model, standardize = "basic")
#> Parameter   | Std. Coef. |   SE |         95% CI | t(28) |      p
#> -----------------------------------------------------------------
#> (Intercept) |       0.00 | 0.00 | [ 0.00,  0.00] | 16.52 | < .001
#> wt          |      -0.82 | 0.10 | [-1.02, -0.62] | -7.89 | < .001

model_parameters(model, standardize = "pseudo")
#> Warning: The following within-group terms have between-group variance:
#>  wt
#> This can inflate standardized within-group parameters associated with
#> these terms. See help("demean", package = "parameters") for modeling
#> between- and within-subject effects.
#> Parameter   | Std. Coef. |   SE |         95% CI | t(28) |      p
#> -----------------------------------------------------------------
#> (Intercept) |            |      |                | 16.52 | < .001
#> wt          |      -1.05 | 0.13 | [-1.31, -0.79] | -7.89 | < .001
```

<sup>Created on 2020-10-20 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mattansb** [comment]
> Ok, will submit insight next week, or are there any things required to make effectsize work?

effectsize works with the current master branch of insight, so I think all is good on that end.

**Comment by mattansb** [comment]
From what I can see, the checks are failing because parameter not depends on a version of insight that is not yet on cran.

Running tests locally, everything seems to work, so... I think my job here is done?

**Pull request merged**